src/debugger/Debugger.cpp

```diff
@@ -206,10 +206,25 @@ bool Debugger::doEval(ExecutionState* state, ByteCodeBlock* byteCodeBlock, uint8
 
     if (type == ESCARGOT_MESSAGE_EVAL_8BIT) {
         str = new Latin1String(data, size);
-    } else {
+    } else if (type == ESCARGOT_MESSAGE_EVAL_16BIT) {
         str = new UTF16String((char16_t*)data, size / 2);
+    } else if (type == ESCARGOT_DEBUGGER_CLIENT_SOURCE_8BIT) {
+        char* sourceNameSrc = (char*)memchr(data, '\0', size);
+        if (sourceNameSrc) {
+            uint32_t sourceNameLen = strlen(data);
```

Comment:
You can compute sourceNameLen as `sourceNameSrc  - data`.
In case of failure, we might consider closing the connection, since that is a communication error.